core/types/l2trace.go

```diff
@@ -148,20 +148,23 @@ type StorageWrapper struct {
 }
 
 type TransactionData struct {
-	Type     uint8           `json:"type"`
-	Nonce    uint64          `json:"nonce"`
-	TxHash   string          `json:"txHash"`
-	Gas      uint64          `json:"gas"`
-	GasPrice *hexutil.Big    `json:"gasPrice"`
-	From     common.Address  `json:"from"`
-	To       *common.Address `json:"to"`
-	ChainId  *hexutil.Big    `json:"chainId"`
-	Value    *hexutil.Big    `json:"value"`
-	Data     string          `json:"data"`
-	IsCreate bool            `json:"isCreate"`
-	V        *hexutil.Big    `json:"v"`
-	R        *hexutil.Big    `json:"r"`
-	S        *hexutil.Big    `json:"s"`
+	Type       uint8           `json:"type"`
+	Nonce      uint64          `json:"nonce"`
+	TxHash     string          `json:"txHash"`
+	Gas        uint64          `json:"gas"`
+	GasPrice   *hexutil.Big    `json:"gasPrice"`
+	GasTipCap  *hexutil.Big    `json:"gasTipCap"`
```

Comment:
then how will you define `GasPrice` here for 1559?   I remember in l1 eth, tx.gas_price for 1559, is real gas price the tx paid then.

Reply:
i think we should.. similar to https://hackmd.io/@timbeiko/1559-json-rpc "gasPrice now interpreted as maxFeePerGas before the tx is mined and 【effectiveGasPrice after it's mined (for EIP-1559 txs)】"

Reply:
`GasPrice` is currently set as `maxFeePerGas` [here](https://github.com/scroll-tech/go-ethereum/blob/develop/core/types/dynamic_fee_tx.go#L92).

My thinking is: What we return here is the fields from the original transaction. For EIP-1559 you should ignore the `GasPrice` field and use `tx.GasTipCap`, `tx.GasFeeCap`, and `header.BaseFee` to calculate the effective gas price.

Does this work for you? Or do you need l2geth to return the effective gas price in this RPC?

Reply:
ok i will post process it.